models/datastore.js

```diff
@@ -0,0 +1,87 @@
+let datastore = require('@google-cloud/datastore');
+let config = require('../config/config');
+
+let client = datastore({
+  projectId: config.projectId,
+  keyFilename: config.googleCloudKey
+});
+
+module.exports.client = client;
+
+module.exports.findAll = (query, successCallback, errorCallback) => {
+  client.runQuery(query, (error, entities) => {
+    if (error) {
+      errorCallback(error);
+    } else {
+      successCallback(entities);
+    }
+  });
+};
+
+module.exports.findOne = (query, successCallback, errorCallback) => {
+  client.runQuery(query.limit(1), (error, entities) => {
+    if (error) {
+      errorCallback(error);
+    } else {
+      successCallback(entities[0]);
+    }
+  });
+};
+
+module.exports.findSupplier = (host, successCallback, errorCallback) => {
+  module.exports.findOne(supplierQuery(host), successCallback, errorCallback);
+};
+
+module.exports.save = (id, kind, dataset, successCallback, errorCallback) => {
+  let key;
+
+  if (id) {
+    key = client.key([kind, parseInt(id, 10)]);
+  } else {
+    key = client.key(kind);
+  }
+
+  let entity = {
+    key: key,
+    data: toDatastore(dataset)
```

Comment:
is it ok that the value to `data:` is an array? In the documentation that I could find on `client#save` data appears to be taking an object only.